File: microservice-architecture/feature-flag/python/test/test_feature_flag.py
import os
import sys
import unittest

sys.path.append(os.getcwd())

from src.feature_flag import FeatureFlag, FeatureFlagService  # noqa: E402


class FeatureFlagTest(unittest.TestCase):
    def test_feature_flag(self) -> None:
        service = FeatureFlagService()
        service.set("new-checkout", FeatureFlag(False, {"user-1": True}))
        self.assertTrue(service.enabled("new-checkout", "user-1"))
        self.assertFalse(service.enabled("new-checkout", "user-2"))


if __name__ == "__main__":
    unittest.main()

File: microservice-architecture/feature-flag/ts/test/test_feature_flag.ts
import { FeatureFlagService } from "../src/feature_flag";

function assertEqual<T>(expected: T, actual: T, message: string): void {
  if (expected !== actual) throw new Error(`${message} expected=${expected} actual=${actual}`);
}

const service = new FeatureFlagService();
service.set("new-checkout", { defaultEnabled: false, allowlist: { "user-1": true } });
assertEqual(true, service.enabled("new-checkout", "user-1"), "allowlisted user");
assertEqual(false, service.enabled("new-checkout", "user-2"), "default disabled");
console.log("feature-flag(ts) tests passed");
